internal/commands/user/output.go

```diff
@@ -26,3 +38,44 @@ func getUserOutputComparerBySuccess(outputs []userOutput) func(i, j int) bool {
 		return outputs[i].err != nil && outputs[j].err == nil
 	}
 }
+
+type userTableRowModifier func(userOutput, map[string]interface{})
+
+func userTableHeaders(authProviderType realm.AuthProviderType) []string {
+	var headers []string
+	switch authProviderType {
+	case realm.AuthProviderTypeAPIKey:
+		headers = append(headers, headerName)
+	case realm.AuthProviderTypeUserPassword:
+		headers = append(headers, headerEmail)
+	}
+	headers = append(
+		headers,
+		headerID,
+		headerType,
+	)
+	return headers
```

Comment:
[nit] we can probably just use `return append(headers, headerID, headerType)` all in one line here